mule-module-apikit/src/main/java/org/mule/module/apikit/validation/RestXmlSchemaValidator.java

```diff
@@ -52,10 +52,24 @@ public void validate(String configId, String schemaPath, MuleEvent muleEvent, IR
             Object input = muleEvent.getMessage().getPayload();
             if (input instanceof InputStream)
             {
-                input = IOUtils.toString((InputStream) input, muleEvent.getMessage().getEncoding());
-                logger.debug("transforming payload to perform XSD validation");
-                DataType<String> dataType = DataTypeFactory.create(String.class, muleEvent.getMessage().getDataType().getMimeType());
-                muleEvent.getMessage().setPayload(input, dataType);
+            	// Do NOT convert to String.
+            	// Either stream is marked/reset or byteArray is copied:
+            	InputStream tempInputStream = (InputStream) input;
+            	ByteArrayOutputStream tempBytes = null;
+            	if (tempInputStream.markSupported()) {
+            		tempInputStream.mark(Integer.MAX_VALUE);
+            		data = loadDocument(tempInputStream);
+            	} else {
+            		tempBytes = new ByteArrayOutputStream();
+					IOUtils.copyLarge(tempInputStream, tempBytes);
+            		data = loadDocument(new ByteArrayInputStream(tempBytes.toByteArray()));
```

Comment:
in the old code, we used to change the input object, so it was catched by the next if. Now that you are not changing it and you are using the data object directly, all the tests that use this function are failing. You should change the if that is next to this one, to a "else if"